datafusion/sql/src/expr/function.rs

```diff
@@ -90,10 +91,30 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
             let partition_by = window
                 .partition_by
                 .into_iter()
+                // ignore window spec PARTITION BY for scalar values
+                .filter(|e| !matches!(e, sqlparser::ast::Expr::Value { .. }))
                 .map(|e| self.sql_expr_to_logical_expr(e, schema, planner_context))
                 .collect::<Result<Vec<_>>>()?;
-            let order_by =
-                self.order_by_to_sort_expr(&window.order_by, schema, planner_context)?;
+
+            let order_by = self.order_by_to_sort_expr(
+                &window
+                    .order_by
+                    .into_iter()
+                    // ignore window spec ORDER BY for scalar values
+                    .filter(|e| {
+                        !matches!(
+                            e,
+                            OrderByExpr {
+                                expr: sqlparser::ast::Expr::Value { .. },
+                                ..
+                            }
+                        )
```

Comment:
its okay, it doesn't fail.... I tried to add a test with pure scalars but it is not stable because of other reasons. I'll decompose the test for now for single rows to prove it doesn't fail

Reply:
weird, it works for 
```
select rank() over (order by 1) from (select 1 a union all select 2 a) q;
```

but fails for 

```
select rank() over (order by null) from (select 1 a union all select 2 a) q;
```

Digging

Reply:
@mustafasrepo can I have your input please. The reason is the query fails

```
select rank() over (order by 1) rnk from (select 1 a union all select 2 a) x
Arrow error: Invalid argument error: number of columns(2) must match number of fields(1) in schema
```

The reason for that is `optimize_projections` removes the unused fields and this breaks the consistency. The question for you,  what do you think is expected behavior for such rare case?

PS. If add a column name to projection or to ORDER BY it will expectedly work.